docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+    }
+  }
+}
+
+const MyJSComponent = React.forwardRef((props, forwardedRef) => {
+  return (
+    <View ref={forwardedRef}>
+      <SubmitButton onSubmit={props.onSubmit} />
+    </View>
+  );
+});
+```
+
+#### Using a getter, (note the addition of `getViewRef`)
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof MyJSComponent>;
+
+  render() {
+    return <MyJSComponent ref={this._captureRef} onSubmit={this._onSubmit} />
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+
+  _onSubmit: () => {
+    if (this._ref != null)
+      this._ref.**getViewRef****()**.measure(() => {});
+    }
+  }
+}
+
+class MyJSComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof View>;
+
+  render() {
+    return (
+      <View ref={this._captureRef}>
+        <SubmitButton onSubmit={props.onSubmit} />
+      </View>
+    );
+  }
+
+  getViewRef(): ?React.ElementRef<typeof View> {
+    return this._ref;
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+}
+```
+
+#### Migrating `.measure(`
+
+Let’s take a look at an example calling `UIManager.measure`. This code might look something like this
+
+```javascript
+if (this._scrollView == null || viewRef == null) {
+  return;
+}
+
+const viewHandle = ReactNative.findNodeHandle(viewRef);
+const scrollViewHandle = ReactNative.findNodeHandle(
+  this._scrollView
+);
+UIManager.measure(scrollViewHandle, (x, y, width, height) => {
+  UIManager.measureLayout(
+    viewHandle,
+    scrollViewHandle,
+    emptyFunction,
+    successCallback
+  );
+});
+```
+
+In order to call `UIManager.measure*` we need to call `findNodeHandle` first and pass in those handles. With the new API, we instead call `measure` directly on native refs without `findNodeHandle`. The example above with the new API looks like this:
+
+```
+if (this._scrollView == null || viewRef == null) {
+  return;
+}
+
+this._scrollView.measure((x, y, width, height) => {
+  viewRef.measureLayout(this._scrollView, successCallback);
+})
+```
+
+`findNodeHandle` can be called with any component as an argument, but the new `.measure*` can only be called on native refs. If the ref originally passed into `findNodeHandle` is not a native ref to start with, use the strategies above in _getting a HostComponent_ to find the native ref.
+
+#### Migrating off `setNativeProps`
+
+setNativeProps will not be supported in the post-Fabric world. To migrate, move all `setNativeProp` values to component state.
+
+**Example**
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _viewRef: ?React.ElementRef<typeof View>;
+
+  render() {
+    const {somePropValue} = this.props;
+    return <View
+       onPress={this._onSubmit}
+       ref={this._captureRef}
+       someProp={somePropValue}
+       style={styles.view} />
+  }
+
+  _captureRef: (ref) => {
+    this._viewRef = ref;
+  }
+
+  _onSubmit: () => {
+    this._viewRef.**setNativeProps****(**{
```

Comment:
Also here, are the asterisks intentional?

Reply:
Nope 👍 Fixed as well.